exchange/gdpr.go

```diff
@@ -11,6 +13,15 @@ import (
 func extractGDPR(bidRequest *openrtb2.BidRequest) (gdpr.Signal, error) {
 	var re regsExt
 	var err error
+
+	if bidRequest.Regs != nil && len(bidRequest.Regs.GPPSID) > 0 {
+		for _, id := range bidRequest.Regs.GPPSID {
+			if id == int8(gppConstants.SectionTCFEU2) {
+				return gdpr.SignalYes, nil
+			}
+		}
+		return gdpr.SignalNo, nil
+	}
```

Comment:
There be dragons.

Should a non-nil GPP SID array be treated the same as a missing one? If there is an array without any elements, might this be interpreted as an explicit `SignalNo`? The len check on line 17 would cause this to fall through and perhaps land as `SignalAmbiguous`.

Reply:
GGPSID is a slice, not a pointer. Is a nil value different from a zero length slice?

Reply:
I would think that for at least as long as we're supporting `regs.ext` as a fallback, the absence of `GPPSID` would be handled differently from a zero length `GPPSID` with the former being considered an ambiguous signal and the latter being an explicit no. If it's ambiguous, we then look in `regs.ext`.
Thoughts?
I poked around the prebid spec and the IAB spec but I don't see any clear direction on this.

Reply:
Yes, GGPSID is a slice. While slices are not technically pointers, their main purpose is to point to a backing array. Therefore, I suppose, if a slice has a nil backing array it is considered nil and `slice == nil` evaluates to `true`.

The problem is I don't know how we should handle the case where a GPP consent string is provided but GPPSID is nil.